### DISCUSSION TITLE
How does `el.time` node work?

### DISCUSSION THREAD

**Discussion by satelllte** [elemaudio/elementary#17] [Q&A] [answered]
So far I haven't found any examples on how to use the [el.time](https://www.elementary.audio/docs/reference/time) node and how it works. So I hope I can get more information on it here 🙏🏻 

Thanks!

**Comment by nick-thompson** [comment, marked as answer]
Yes of course! The time node (only available in the web-renderer and offline-renderer) outputs a signal that is very literally a monotonically increasing count of elapsed samples that the Elementary engine has processed. This allows you to coordinate various signals i.e. "time sync" or "host sync."

It's quite a simple utility; but you can derive useful data out of it. For example, if you know that the current time is T (based on el.time) then you can use your own BPM and time signature information (if you have it; or if it applies to your application) to calculate a `ppqn` or `ppq` signal like you might find in a DAW or in a MIDI file. That signal could tell you where you are within the current beat or bar of the song, from which you could derive a quarter-note-synced LFO that runs perfectly in time with a half-note-synced LFO, etc. Does that help?

**Reply by satelllte** [reply]
Yes, I guess I'm getting now how it works. Thanks! 

So just to clarify, for example in a web enviroment, it increases by `AudioContext.sampleRate` value each second, is it correct?

**Reply by nick-thompson** [reply]
Yes exactly
